easybuild/easyblocks/l/lua.py

```diff
@@ -0,0 +1,85 @@
+##
+# Copyright 2009-2018 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for software that uses the GNU installation procedure,
+i.e. configure/make/make install, implemented as an easyblock.
+
+@author: Stijn De Weirdt (Ghent University)
+@author: Dries Verdegem (Ghent University)
+@author: Kenneth Hoste (Ghent University)
+@author: Pieter De Baets (Ghent University)
+@author: Jens Timmerman (Ghent University)
+@author: Toon Willems (Ghent University)
+"""
+import os
+
+from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import apply_regex_substitutions
+from easybuild.tools.run import run_cmd
+
+
+class EB_Lua(ConfigureMake):
+    """
+    Support for building and installing Lua
+    """
+
+    def configure_step(self, cmd_prefix=''):
+        """
+        Configure step
+
+        Lua does not need a configure step. In this step we just patch the
+        `luaconf.h` file in the sources to point to the correct Lua Root
+        """
+        src_dir = self.src[0]['finalpath']
+        luaconf_h = os.path.join(src_dir, 'src', 'luaconf.h')
+
+        # Open file
+        self.log.debug("Patching luaconf.h at {}".format(luaconf_h))
+        installdir = os.path.join(self.installdir, '')  # Add trailing slash
+        apply_regex_substitutions(luaconf_h, [(r'/usr/local/', installdir)])
+
+    def test_step(self):
+        out = super(EB_Lua, self).test_step()
+
+        # Check if the path where lua looks for packages is correctly pointing
+        # to installdir
+        cmd = "src/lua -e 'io.write(package.path)'"
```

Comment:
This should be a sanity check command, so we check the `lua` binary that actually gets installed?

When writing a custom easyblock, you should include a custom implementation of `sanity_check_step` anyway...